src/qibolab/instruments/qblox/acquisition.py

```diff
@@ -0,0 +1,144 @@
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+import numpy as np
+
+SAMPLING_RATE = 1e9
+
+
+def demodulate(input_i, input_q, frequency):
+    """Demodulates and integrates the acquired pulse."""
+    # DOWN Conversion
+    # qblox does not remove the offsets in hardware
+    modulated_i = input_i - np.mean(input_i)
+    modulated_q = input_q - np.mean(input_q)
+
+    num_samples = modulated_i.shape[0]
+    time = np.arange(num_samples) / SAMPLING_RATE
+    cosalpha = np.cos(2 * np.pi * frequency * time)
+    sinalpha = np.sin(2 * np.pi * frequency * time)
```

Comment:
```suggestion
    time = np.arange(num_samples) 
    phase = (2 * np.pi * frequency / SAMPLING_RATE) * time
    cosalpha = np.cos(phase)
    sinalpha = np.sin(phase)
```

More efficient and more readable.

(keeping `time` at the end will execute the array operation only once)